### PR TITLE
voucher cancellation flow

### PR DESCRIPTION
**Pull request by twrichards**
### Notable changes from other cancellations flow...

- Notion of an _**'escalation route'**_ which...
  - displays extra paragraph on the review/feedback stage
  - when the user clicks to `Confirm cancellation`, instead of a cancellation call to `members-data-api` to perform the cancellation, there's just a call to update the SF Case with...
    - Priority : `High`
    - Status: `New`
    - Subject: `Online Cancellation MANUAL INTERVENTION REQUIRED - ` plus the escalation cause(s)
    - _NOTE the queue assignment is handled in Salesforce (as per usual) - see https://github.com/guardian/salesforce/pull/204_
  - appropriate message on the confirmation page (that customer services will be in touch to complete the cancellation)

<br/>

- 'cancellation policy' selector on the first page (below the reason radios) to choose cancellation effective date, choices are 'today' or the 'end of the current billed period' (which is populated using newly exposed `chargedThroughDate` - see https://github.com/guardian/members-data-api/pull/434). **If the user selects 'Today' then that puts them on the _'escalation route'_.**

- check for 'outstanding holiday stop credits' using the dedicated endpoint for this (see [holiday-stop-api/cancel GET](https://github.com/guardian/support-service-lambdas/tree/master/handlers/holiday-stop-api#get-stagehsrsubscription_namecanceleffectivecancellationdateyyyy-mm-dd) - same as CSR cancellation flow in SF). **If there are any then that puts them in the _'escalation route'_.** This has been implemented such that it's easy to add a similar check for 'delivery problem credits' when we do GW cancellation flow.

### Subscriptions page
_see bit with red outline_
image

### Step 1
_notice the additional radios at the bottom for choosing effective date_
image
_the paragraph with red outline only appears for non-Print+ subscriptions, since Print+ already pay for digipack access_

#### Normal (having chosen `I am unhappy with Guardian journalism` and the end of 'billing period' option)
image

#### _Escalation route_ (having chosen `I am unhappy with Guardian journalism` and the `Today` option)
image
_the bit in red is the bit that appears if escalated_

### Normal
_you can see the sub is cancelled like normal_
image

#### Escalation route_ (the sub also had outstanding holiday stops in addition to having selected `Today`)
image

### Case(s) in Salesforce
_notice the difference in subject, owner, priority and status between the 'Normal' and 'Escalated'_
image
_the escalated one more closely..._
image

### CODE REVIEW

File: app/client/components/cancel/cancellationReason.ts [hunk 1]
```diff
@@ -14,6 +13,9 @@ export interface CancellationReason {
 }
 
 export type CancellationReasonId =
+  | "mma_covid"
+  | "mma_delivery_issue"
+  | "mma_redemption_issue"
```
Comment (twrichards) [inline app/client/components/cancel/cancellationReason.ts:18]: Need to add in Zuora...

- [ ] DEV
- [ ] UAT
- [ ] PROD

File: app/client/components/cancel/membership/membershipCancellationFlowStart.tsx [hunk 2]
```diff
@@ -68,63 +64,19 @@ export const membershipCancellationFlowStart = (
...
-                  paddingTop: "5px"
-                }}
-              >
-                Free access to the premium tier of the Guardian app -{" "}
-                {clickHereToFindOutMoreAboutOurNewFeatures}
-              </li>
-            </ul>
-          )
-        }}
-      </GoogleOptimiseAwaitFlagWrapper>
```
Comment (twrichards) [inline app/client/components/cancel/membership/membershipCancellationFlowStart.tsx:127]: just come cleanup of some old un-used stuff

File: app/client/components/cancel/membership/membershipCancellationReasons.tsx [hunk 3]
```diff
@@ -10,27 +9,7 @@ export const membershipCancellationReasons: CancellationReason[] = [
...
-            }}
-          >
-            For as little as £2 you can continue to support the Guardian - and
-            it only takes a minute.
-          </span>
-        </div>
-        <SwitchToContributionPlaceholder />
-      </>
-    ),
-    experimentTriggerFlag: "showSwitchToContributionPlaceholder"
```
Comment (twrichards) [inline app/client/components/cancel/membership/membershipCancellationReasons.tsx:33]: some more cleanup of un-used stuff

**Comment by twrichards** [comment]
@FYI @JessL123

File: app/client/components/cancel/voucher/voucherCancellationFlowStart.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    eventCategory: "cancellation",
+    eventAction: "click",
+    eventLabel
+  });
+
+export const voucherCancellationFlowStart = (subscription: Subscription) => {
+  const mainPlan = getMainPlan(subscription);
+
+  const isEligibleForFreeDigipackAccess =
+    mainPlan?.name?.indexOf("plus Digi") === -1;
```
Comment (rBangay) [inline app/client/components/cancel/voucher/voucherCancellationFlowStart.tsx:18]: `!mainPlan?.name.includes("plus Digi")`

Comment (twrichards) [reply app/client/components/cancel/voucher/voucherCancellationFlowStart.tsx:18]: wasn't sure if `includes` has babel issues

Comment (rBangay) [reply app/client/components/cancel/voucher/voucherCancellationFlowStart.tsx:18]: you're correct, ignore this. Babel transpiles ES6 code but doesn't polyfill ES6 methods

**Pull request merged**